component/booksCard.jsx

```diff
@@ -0,0 +1,57 @@
+import useStore from '../hooks/useStore';
+import { useEffect } from 'react';
+import styled from 'styled-components';
+
+const Api = process.env.NEXT_PUBLIC_REACT_APP_API_KEY;
+
+export default function BooksCard({ apikey }) {
+  const fetchApi = useStore(state => state.fetchApi);
+  const fetchedData = useStore(state => state.fetchedData);
+
+  useEffect(() => {
+    fetchApi(
+      `https://api.nytimes.com/svc/books/v3/lists/current/hardcover-fiction.json?api-key=${apikey}`
+    );
+  }, [fetchApi]);
+
+  return (
+    <CardWrapper>
+      {fetchedData?.results?.books !== undefined ? (
+        fetchedData.results.books.map((book, index) => {
+          return (
+            <StyledCard key={index}>
```

Comment:
Using the `index` is not a good practise and could lead to errors. Maybe the `book` object has an `id` proberty you could use here. If not, `index` is a doable workaround.